website/source/docs/auth/jwt.html.md

```diff
@@ -4,17 +4,152 @@ page_title: "JWT - Auth Methods"
+
+
+### Claim specifications and JSON Pointer
+
+Some parameters (e.g. `bound_claims` and `groups_claim`) are used to point to data within the JWT. If
+the desired key is at the top of level of the JWT, the name can be provided directly. If it is nested at a
+lower level, a JSON Pointer may be used.
+
+Assume the following JSON data to be referenced:
+
+```json
+{
+  "division": "North America",
+  "groups": {
+    "primary": "Engineering",
+    "secondary": "Software"
+  }
+}
+```
+
+A parameter of `"division"` will reference "North America", as this is a top level key. A parameter
+`"/groups/primary"` uses JSON Pointer syntax to reference "Engineering" at a lower level. Any valid
+JSON Pointer can be used as a selector. Refer to the
+[JSON Pointer RFC](https://tools.ietf.org/html/rfc6901) for a full description of the syntax
+
+
+## OIDC Authentication
+
+This section covers the setup and use of OIDC roles. If a JWT is to be provided directly,
+refer to the [JWT Authentication](/docs/auth/jwt.html#jwt-authentication) section below. Basic
+familiarity with [OIDC concepts] (https://developer.okta.com/blog/2017/07/25/oidc-primer-part-1)
+is assumed.
+
+Vault includes two built-in OIDC login flows: the Vault UI, and the CLI
+using a `vault login`.
+
+### Redirect URIs
+
+An important part of OIDC role configuration is properly setting redirect URIs. This must be
+done both in Vault and with the OIDC provider, and these configurations must align. The
+redirect URIs are specified for a role with the `allowed_redirect_uris` parameter. There are
+different redirect URIs to configure the Vault UI and CLI flows, so one or both will need to
+be set up depending on the installation.
+
+**CLI**
+
+If you plan to support authentication via `vault login -method=oidc`, a localhost redirect URI
+must be set. This can usually be: `http://localhost:8300/oidc/callback`. Logins via the CLI may
+specify a different listening port if needed, and a URI with this port must match one of the
+configured redirected URIs. These same "localhost" URIs must be added to the provider as well.
+
+**Vault UI**
+
+Logging in via the Vault UI requires a redirect URI of the form:
+`https://{host:port}/ui/vault/auth/{path}/oidc/callback`
+
+The "host:port" must be correct for the Vault server, and "path" must match the path the JWT
+backend is mounted at (e.g. "oidc" or "jwt").
+If [namespaces](https://www.vaultproject.io/docs/enterprise/namespaces/index.html) are being used,
+they must be added as query parameters, for example:
+
+`https://vault.example.com:8200/ui/vault/auth/oidc/oidc/callback?namespace=my_ns`
+
+### OIDC Login (Vault UI)
+
+1. Select the "OIDC" login method.
+1. Enter a role name if necessary.
+1. Press "Sign In" and complete the authentication with the configured provider.
+
+### OIDC Login (CLI)
+
+The CLI login defaults to path of `/oidc`. If this auth method was enabled at a
+different path, specify `-path=/my-path` in the CLI.
```

Comment:
Do we want to mention the redirect_uri format for this as well? Also is there a way to specify a port other than 8300?

Reply:
I mention the redirect format and port option for the CLI in the redirect section. I'll bring up the port again here, but redirect got a lot words in the other section.

Reply:
Aha, I think I was looking for something like the UI's patterned url, but going back and re-reading it seems that the CLI will just use whatever you set for `allowed_redirect_uris`, including the port?